src/ol_infrastructure/substructure/keycloak/__main__.py

```diff
@@ -489,6 +519,15 @@
                 realm_name=realm_name,
             ).apply(json.dumps),
         )
+        if len(client_detail) > 1:
+            for role in client_detail[1:]:
```

Comment:
You can leave out the conditional, because if the list is only length 1 then `[1:]` will yield an empty list, short-circuiting the `for` loop.